sdk/monitor/azure-monitor-opentelemetry-autoconfigure/src/main/java/com/azure/monitor/opentelemetry/autoconfigure/implementation/quickpulse/QuickPulseCoordinator.java

```diff
@@ -123,15 +139,30 @@ private long ping() {
         return 0;
     }
 
-    private void handleReceivedHeaders(QuickPulseHeaderInfo currentQuickPulseHeaderInfo) {
-        String redirectLink = currentQuickPulseHeaderInfo.getQpsServiceEndpointRedirect();
+    private QuickPulseStatus handleReceivedPingHeaders(IsSubscribedHeaders pingHeaders) {
+        String redirectLink = pingHeaders.getXMsQpsServiceEndpointRedirectV2();
         if (!Strings.isNullOrEmpty(redirectLink)) {
             qpsServiceRedirectedEndpoint = redirectLink;
         }
 
-        long newPollingInterval = currentQuickPulseHeaderInfo.getQpsServicePollingInterval();
-        if (newPollingInterval > 0) {
-            qpsServicePollingIntervalHintMillis = newPollingInterval;
+        String pollingIntervalHeader = pingHeaders.getXMsQpsServicePollingIntervalHint();
```

Comment:
Why the logic has changed?

Coud you please add a comment or do an extract method with a self-explanatory method name?